api/src/user/controllers/role.controller.ts

```diff
@@ -142,12 +147,29 @@ export class RoleController extends BaseController<Role, RoleStub> {
   @CsrfCheck(true)
   @Delete(':id')
   @HttpCode(204)
-  async deleteOne(@Param('id') id: string) {
-    const result = await this.roleService.deleteOne(id);
-    if (result.deletedCount === 0) {
-      this.logger.warn(`Unable to delete Role by id ${id}`);
-      throw new NotFoundException(`Role with ID ${id} not found`);
+  async deleteOne(@Param('id') id: string, @Session() session: ExpressSession) {
+    const currentUser = await this.userService.findOneAndPopulate(
+      session.passport.user.id,
+      ['roles'],
+    );
+    if (!currentUser) {
+      throw new NotFoundException('User not found');
+    }
+
+    const roles = currentUser.roles.map((role) => role.id);
+
+    if (roles.includes(id)) {
+      throw new ForbiddenException("Your account's role can't be deleted");
+    } else {
+      try {
+        const result = await this.roleService.deleteOne(id);
+        if (result.deletedCount === 0) {
+          throw new NotFoundException(`Role with ID ${id} not found`);
+        }
+        return result;
+      } catch (error) {
+        throw new NotFoundException(`Role with ID ${id} not found`);
+      }
     }
```

Comment:
This code can be made cleaner and more concise:
```suggestion
   if (roles.includes(id)) {
      throw new ForbiddenException("Your account's role can't be deleted");
    }
    const result = await this.roleService.deleteOne(id);
    if (result.deletedCount === 0) {
      throw new NotFoundException(`Role with ID ${id} not found`);
    }
    return result;
   }
```

Reply:
You can maybe add a  `this.logger.warn(`Unable to delete Role by id ${id}`);` 
before throwing the error